assembly/src/assembler/context.rs

```diff
@@ -353,11 +355,12 @@ impl ModuleContext {
         // make sure a procedure with this name as not been compiled yet and is also not currently
         // on the stack of procedures being compiled
         if self.compiled_procs.iter().any(|p| p.label() == name)
-            || self.proc_stack.iter().any(|p| p.name == name)
+            || self.proc_stack.iter().any(|p| p.name.as_str() == name)
```

Comment:
We don't have a safe/stable way to declare a `const String` because string is a growable buffer. This could instead be moved as constant of the procedure name instead of a loose constant